src/module.c

```diff
@@ -661,12 +689,13 @@ void moduleReleaseTempClient(client *c) {
     c->flags = CLIENT_MODULE;
     c->user = NULL; /* Root user */
     c->cmd = c->lastcmd = c->realcmd = NULL;
-    if (c->bstate.async_rm_call_handle) {
-        RedisModuleAsyncRMCallPromise *promise = c->bstate.async_rm_call_handle;
-        promise->c = NULL; /* Remove the client from the promise so it will no longer be possible to abort it. */
-        freeRedisModuleAsyncRMCallPromise(promise);
-        c->bstate.async_rm_call_handle = NULL;
-    }
+
+    /* reset client stat that can persist */
+    pubsubUnsubscribeAllChannels(c,0);
+    pubsubUnsubscribeShardAllChannels(c, 0);
+    pubsubUnsubscribeAllPatterns(c,0);
```

Comment:
```suggestion
    pubsubUnsubscribeAllChannels(c, 0);
    pubsubUnsubscribeShardAllChannels(c, 0);
    pubsubUnsubscribeAllPatterns(c, 0);
```
not that i care much, but to avoid future changes like you (maybe unintentionally) did above
```diff
-        moduleTempClientCap = moduleTempClientCap ? moduleTempClientCap*2 : 32;
+        moduleTempClientCap = moduleTempClientCap ? moduleTempClientCap * 2 : 32;
```

Reply:
not sure how that happened unless editor reformatting it on me.

Reply:
i didn't mind that (since it's better), son't mind either way.
but the new code you added (calls to pubsubUnsubscribeAllChannels and pubsubUnsubscribeAllPatterns) should better be fixed.

Reply:
ok, fixed pubSub stuff. that was a copy/paste from networking.c should I fix it there too?

Reply:
nahh, just avoid in new code.